### PR TITLE
Make authorization errors more specific

### PR DESCRIPTION
**Pull request by nikhil-stytch**
Define specific errors for local authorization

### CODE REVIEW

File: stytch/shared/rbac_local.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func PerformAuthorizationCheck(
 		}
 	}
 
-	return stytcherror.NewClientLibraryError("Member is not authorized")
+	return stytcherror.NewSessionAuthorizationError()
```
Comment (logan-stytch) [inline stytch/shared/rbac_local.go:36]: [dust] Can we call this a `NewPermissionError()` or `NewRBACPermissionError()` for parity with other SDKs?

File: stytch/shared/rbac_local.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ func PerformAuthorizationCheck(
 	}
 
 	if subjectOrgID != authorizationCheck.OrganizationID {
-		return stytcherror.NewClientLibraryError("Subject organization ID does not match ID from request")
+		return stytcherror.NewSessionAuthorizationTenancyError()
```
Comment (logan-stytch) [inline stytch/shared/rbac_local.go:20]: [sand] bonus if we can take `subjectOrgID` and `authorizationCheck.OrganizationID` as params for the constructor to display a helpful error message

**Review by logan-stytch — APPROVED** [review]

File: stytch/shared/rbac_local.go (hunk 1 above)
Comment (nikhil-stytch) [reply stytch/shared/rbac_local.go]: [5c737e0](https://github.com/stytchauth/stytch-go/pull/159/commits/5c737e045a12826dcc65a62e80b8f4e6428b88ac)

File: stytch/shared/rbac_local.go (hunk 2 above)
Comment (nikhil-stytch) [reply stytch/shared/rbac_local.go]: [5c737e0](https://github.com/stytchauth/stytch-go/pull/159/commits/5c737e045a12826dcc65a62e80b8f4e6428b88ac)

File: stytch/stytcherror/stytcherror.go [hunk 3]
```diff
@@ -77,4 +77,24 @@ func NewClientLibraryError(message string) error {
 	}
 }
 
+func NewSessionAuthorizationTenancyError(expectedOrgID string, actualOrgID string) error {
+	msg := fmt.Sprintf("OrganizationID in AuthorizationCheck request does not match session's Organization ID. Got %v but expected %v", actualOrgID, expectedOrgID)
```
Comment (logan-stytch) [inline stytch/stytcherror/stytcherror.go:81]: Probably want `%q` or `%s` for formatting

File: stytch/shared/rbac_local.go [hunk 4]
```diff
@@ -17,7 +17,7 @@ func PerformAuthorizationCheck(
 	}
 
 	if subjectOrgID != authorizationCheck.OrganizationID {
-		return stytcherror.NewClientLibraryError("Subject organization ID does not match ID from request")
+		return stytcherror.NewSessionAuthorizationTenancyError(subjectOrgID, authorizationCheck.OrganizationID)
```
Comment (logan-stytch) [inline stytch/shared/rbac_local.go:20]: The constructor seems to expect authCheck.orgID first -- or maybe it's just not clear which is "expected"
The Ruby error makes this explicit:
```rb
msg = "Subject organization_id #{subject_org_id} does not match authZ request organization_id #{request_org_id}"
```

**Pull request merged**